logstash-core/lib/logstash/runner.rb

```diff
@@ -184,6 +186,20 @@ def configure_plugin_paths(paths)
     end
   end
 
+  def agent_class
+    return @agent_class if @agent_class # We don't want to load these things twice
+
+    LogStash::AgentPluginManager.load_all
```

Comment:
I would probably break this function into more parts. Specially having this ```load_all``` as something done probably in construction time of this class. What do you think?